Core/Src/monitor.c

```diff
@@ -209,236 +70,185 @@ uint16_t adjust_pedal_val(uint32_t raw, int32_t offset, int32_t max)
 		       (uint16_t)(raw - offset) * 100 / (max - offset);
 }
 
-void vPedalsMonitor(void *pv_params)
+/**
+ * @brief Callback for pedal fault debouncing.
+ * 
+ * @param arg The fault message as a char*.
+ */
+void pedal_fault_cb(void *arg)
 {
-	const uint8_t num_samples = 10;
-	enum { ACCELPIN_2, ACCELPIN_1, BRAKEPIN_1, BRAKEPIN_2 };
-
-	static pedals_t sensor_data;
 	fault_data_t fault_data = { .id = ONBOARD_PEDAL_FAULT,
 				    .severity = DEFCON1 };
+	fault_data.diag = (char *)arg;
+	queue_fault(&fault_data);
+}
+
+/**
+ * @brief Function to send raw pedal data over CAN.
+ * 
+ * @param arg A pointer to an array of 4 unsigned 32 bit integers.
+ */
+void send_pedal_data(void *arg)
+{
+	static const uint8_t adc_data_size = 4;
+	uint32_t adc_data[adc_data_size];
+	/* Copy contents of adc_data to new location in memory because we endian swap them */
+	memcpy(adc_data, (uint32_t *)arg, adc_data_size * sizeof(adc_data[0]));
+
 	can_msg_t accel_pedals_msg = { .id = CANID_PEDALS_ACCEL_MSG,
 				       .len = 8,
 				       .data = { 0 } };
 	can_msg_t brake_pedals_msg = { .id = CANID_PEDALS_BRAKE_MSG,
 				       .len = 8,
 				       .data = { 0 } };
+
+	endian_swap(&adc_data[ACCELPIN_1], sizeof(adc_data[ACCELPIN_1]));
+	endian_swap(&adc_data[ACCELPIN_2], sizeof(adc_data[ACCELPIN_2]));
+	memcpy(accel_pedals_msg.data, &adc_data, accel_pedals_msg.len);
+	queue_can_msg(accel_pedals_msg);
+
+	endian_swap(&adc_data[BRAKEPIN_1], sizeof(adc_data[BRAKEPIN_1]));
+	endian_swap(&adc_data[BRAKEPIN_2], sizeof(adc_data[BRAKEPIN_2]));
+	memcpy(brake_pedals_msg.data, adc_data + 2, brake_pedals_msg.len);
+	queue_can_msg(brake_pedals_msg);
+}
+
+osThreadId_t pedals_monitor_handle;
+const osThreadAttr_t pedals_monitor_attributes = {
+	.name = "PedalMonitor",
+	.stack_size = 128 * 8,
+	.priority = (osPriority_t)osPriorityRealtime,
+};
+
+void vPedalsMonitor(void *pv_params)
+{
 	uint32_t adc_data[4];
-	bool is_braking = false;
+	osTimerId_t send_pedal_data_timer =
```

Comment:
may not be the most memory-safe but seems to be a pretty awesome way of handling periodic sending of pedal data.